### PR TITLE
[PPL-197] Fixed shows incorrect info for the tables/dbs/etc with dash in the …

### PR DESCRIPTION
**Pull request by askomorokhov**
…names.

### CODE REVIEW

**Comment by daniel-nichter** [comment]
@askomorokhov What case is this trying to fix? There's no test case so I can't see what the intent is. Also, this breaks the code: add this test case and it will panic:
```go
        q = "select `col` from `table-1` where `id` = 5"
        t.Check(
                query.Fingerprint(q),
                Equals,
                "select `col` from `table-1` where `id` = ?",
        )
```
This code should have 100% test cover, as a note in `query_test.go` mentions:
```
func (s *TestSuite) SetUpSuite(t *C) {
        // Uncomment to check for 100% test coverage:
        // query.Debug = true
}
```

File: query/query.go [hunk 1]
```diff
@@ -112,6 +113,7 @@ var stateName map[byte]string = map[byte]string{
 	15: "orderBy",
 	16: "onDupeKeyUpdate",
 	17: "inNumberInWord",
+	18: "inBackticks",
```
Comment (percona-csalguero) [inline query/query.go]: In the above definition at line 83 `inBackTicks` = iota + 7 = 0 + 7 = 7.
Is it correct to put it at position 18 here?

**Comment by percona-csalguero** [comment]
Hi @daniel-nichter 
Just a question about the proposed test case. 
In other tests cases I see, for example, this:
```
     q = "select foo_1 from foo_2_3"
     t.Check(
         query.Fingerprint(q),
         Equals,
         "select foo_? from foo_?_?",
     )
```
and that makes me think  that in the test you proposed, we should check for `table-?` instead of `table-1`.
Is this correct?
Thanks

**Comment by percona-csalguero** [comment]
I think this commit should be reverted as it breaks all the tests for the query package.
The error is not in the query fingerprinter, it is here:

https://github.com/percona/qan-agent/blob/master/query/mysql/transform.go#L54-L77
```
func DMLToSelect(query string) string {
	m := updateRe.FindStringSubmatch(query)
	// > 2 because we need at least a table name and a list of fields
	if len(m) > 2 {
		return updateToSelect(m)
	}

	m = deleteRe.FindStringSubmatch(query)
	if len(m) > 1 {
		return deleteToSelect(m)
	}

	m = insertRe.FindStringSubmatch(query)
	if len(m) > 2 {
		return insertToSelect(m)
	}

	m = insertSetRe.FindStringSubmatch(query)
	if len(m) > 2 {
		return insertWithSetToSelect(m)
	}

	return ""
}
```

And to be more precise, the problem PPL-197 is trying to solve is in this regex:
https://github.com/percona/qan-agent/blob/master/query/mysql/transform.go#L30
```
	insertRe    = regexp.MustCompile(`(?i)^(?:insert(?:\s+ignore)?|replace)\s+.*?\binto\s+(.*?)\(([^\)]+)\)\s*values?\s*\((.*?)\)\s*(?:\slimit\s|on\s+duplicate\s+key.*)?\s*$`)
```

**Comment by askomorokhov** [comment]
> @askomorokhov What case is this trying to fix?

@daniel-nichter, this hot-fix is trying to generate a correct finger-print for queries like this:
```sql
SELECT BENCHMARK(100000000, pow(rand(), rand())), name from `-hj-7d6-%$*()!-7jd-kf-g9+88h-`.`-aaahj-7d6-shdj5-7jd-kf-g988h-`;
```

**Pull request merged**